shared/js/template.js

```diff
@@ -113,25 +113,43 @@
    *                          - safe, a list of properties that contain
    *                          HTML that is known and are
    *                          "known" to ignore.
-   */
-  Template.prototype.interpolate = function(data, options) {
-    // This _should_ be rewritten to use Firefox's support for ES6
-    // default parameters:
-    // ... = function(data, options = { safe: [] }) {
-    //
-    options = options || {};
-    options.safe = options.safe || [];
+  */
+  Template.prototype.interpolate = function(data = {}, options = { safe: [] }) {
+    var safeList = options.safe;
+    var hasSafeList = Boolean(safeList);
+
+    if (hasSafeList && !Array.isArray(safeList)) {
+      throw new TypeError(
+        `"options.safe" must be an Array, not ${typeof options.safe}`
+      );
+    }
+
+    var hasSafeProperties = hasSafeList && safeList.length;
```

Comment:
Actually I did not really change it. The empty Array is still in the default parameters. I just throw an error if the type of `safe` is forced to something else than an Array.